homeassistant/components/binary_sensor/mqtt.py

```diff
@@ -80,3 +83,9 @@ def name(self):
     def is_on(self):
         """True if the binary sensor is on."""
         return self._state
+
+    @property
+    def sensor_class(self):
+        """Return the class of this sensor."""
+        return self._sensor_class if self._sensor_class in \
+                                     SENSOR_CLASSES else None
```

Comment:
In my other components, I did this check during setup, so that it was obvious if you choose something that isn't supported. This will just silently ignore invalid values. Maybe at least log something at warn level?